build.gradle

```diff
@@ -142,24 +142,19 @@ String artifactVersionsApi = "https://artifacts-api.elastic.co/v1/versions"
 
 tasks.register("configureArchitecture") {
     String arch = System.properties['os.arch']
-    String osName = System.properties['os.name']
     String beatsArch = arch
     String esArch = arch
+    String osName = (System.properties['os.name'] ==~ /Mac OS X/) ? "darwin" : "linux"
```

Comment:
I think with this, it breaks  on Windows

Reply:
This line is the simplification of L158~162L and I think we need to separately visit Windows case. I have a doubt that even if we fix OS name, artifact extensions (such as `.msi`) is not properly set in this logic.

Reply:
We recommend using Docker on Windows: [guide link](https://github.com/elastic/logstash/blob/main/qa/integration/README.md#testing-on-windows)

Reply:
Now I see, you are right.